src/utils.ts

```diff
@@ -19,6 +30,51 @@ class Utils {
     );
   }
 
+  /**
+   * Encode the given classic address and tag into an x-address.
+   *
+   * @see https://xrpaddress.info/
+   *
+   * @param classicAddress A classic address to encode.
+   * @param tag An optional tag to encode.
+   * @returns A new x-address if inputs were valid, otherwise undefined.
+   */
+  public static encodeXAddress(
+    classicAddress: string,
+    tag: number | undefined
```

Comment:
instead of defining it as `tag: number | undefined` should we instead use the optional parameter syntax of `tag?: number`?

Reply:
This is cool, I did not realize you could mark params as optional this way. 

I think this is a poor choice for this particular API because it means that TS programmers can unintentionally omit this param. I think that all inputs to an X-address should be explicitly stated, since X-addresses are specifically trying to solve the problem of people leaving off tags. 

If you feel this is a strong anti-pattern in TS / JS LMK and I don't care too much .

Reply:
Interesting, I think it is okay to do `number | undefined` but `tag?: number` is the same thing technically.

Why is undefined an option if we are forcing people to always have a tag?

Is there a default value for destination tag?

Reply:
NVM talked on slack I see the reasoning now